pkgs/http/lib/src/browser_client.dart

```diff
@@ -15,7 +15,13 @@ import 'streamed_response.dart';
 /// Create a [BrowserClient].
 ///
 /// Used from conditional imports, matches the definition in `client_stub.dart`.
-BaseClient createClient() => BrowserClient();
+BaseClient createClient() {
+  if (const bool.fromEnvironment('no_default_http_client')) {
```

Comment:
Aren't these often SCREAMING_FOR_SOME_REASON?

No need for a foolish consistency, just checking.

Reply:
After a quick survey, they don't seem to be.